code/modules/mob/living/simple_mob/subtypes/animal/alien animals/stardog.dm

```diff
@@ -0,0 +1,1730 @@
+//I'm sorry for this file, no one should have to deal with this
+//I am in a pain trance and coding is the only thing that can distract me
+//I am a dwarf in a fey mood, but what I make will not be a masterwork, woe
+
+/datum/category_item/catalogue/fauna/stardog
+	name = "Alien Wildlife - Star Dog"
+	desc = "I appears to be a canine of some sort, though absolutely massive in scale and surrounded in radical redspace energies!"
+	value = CATALOGUER_REWARD_SUPERHARD
+
+/mob/living/simple_mob/vore/overmap/stardog
+	name = "dog"
+	desc = "It is a relatively ordinary looking canine mutt! It radiates mischief and otherworldly energy..."
+	tt_desc = "E Canis lupus stellarus"
+
+	scanner_desc = "I appears to be a canine of some sort, though absolutely massive in scale and surrounded in radical redspace energies!"
+	catalogue_data = list(/datum/category_item/catalogue/fauna/stardog)
+
+	icon = 'icons/mob/vore.dmi'
+	icon_state = "woof"
+	icon_living = "woof"
+	icon_dead = "woof_dead"
+	icon_rest = "woof_rest"
+
+	om_child_type = /obj/effect/overmap/visitable/ship/simplemob/stardog
+
+	response_help = "pets"
+	response_disarm = "rudely paps"
+	response_harm = "punches"
+
+	attacktext = list("nipped", "chomped", "bullied", "gnaws on")
+	attack_sound = 'sound/voice/bork.ogg'
+	friendly = list("snoofs", "nuzzles", "ruffs happily at", "smooshes on")
+
+	ai_holder_type = /datum/ai_holder/simple_mob/woof/stardog
+
+	has_langs = list(LANGUAGE_ANIMAL, LANGUAGE_CANILUNZT, LANGUAGE_GALCOM)
+	say_list_type = /datum/say_list/softdog
+	swallowTime = 0.1 SECONDS
+
+	loot_list = list(/obj/random/underdark/uncertain)
+
+	armor = list(
+		"melee" = 1000,
+		"bullet" = 1000,
+		"laser" = 1000,
+		"energy" = 1000,
+		"bomb" = 1000,
+		"bio" = 1000,
+		"rad" = 1000)
+
+	armor_soak = list(
+		"melee" = 1000,
+		"bullet" = 1000,
+		"laser" = 1000,
+		"energy" = 1000,
+		"bomb" = 1000,
+		"bio" = 1000,
+		"rad" = 1000
+		)
+
+	movement_cooldown = 5
+	copy_prefs_to_mob = FALSE
+	player_msg = "The dog accepts you into itself, allowing you to dictate what will happen. The dog occasionally thinks unknowable thoughts, though you can understand some of its needs and desires. The dog shares its experience with you. You can navigate space, 'transition' to certain locations, and you can dine upon some of the space weather. The dog doesn't seem to know how any of this works exactly, this is just how things are for the dog, they come as naturally to the dog as blinking."
+
+	var/affinity = 0
+	var/obj/structure/control_pod/control_node = null
+	var/shipvore = FALSE	//Enable this to allow the star dog to eat spaceships by dragging them onto its sprite.
```

Comment:
I think this should be a general /mob/living/simple_mob/vore/overmap var, rather than stardog-specific